src/Ng/CommandHelpers.cs

```diff
@@ -153,167 +245,150 @@ public static int GetInterval(IDictionary<string, string> arguments)
+                string storageBaseAddressStr;
+                TryGetArgument(arguments, StorageBaseAddress, out storageBaseAddressStr);
+                if (!string.IsNullOrEmpty(storageBaseAddressStr))
+                {
+                    storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
 
-            string storageKeyValue;
-            if (!arguments.TryGetValue("-compressedStorageKeyValue", out storageKeyValue))
-            {
-                TraceRequiredArgument("-compressedStorageKeyValue");
-                return null;
-            }
+                    storageBaseAddress = new Uri(storageBaseAddressStr);
+                }
 
-            string storageContainer;
-            if (!arguments.TryGetValue("-compressedStorageContainer", out storageContainer))
-            {
-                TraceRequiredArgument("-compressedStorageContainer");
-                return null;
-            }
+                string storageAccountName;
+                TryGetArgument(arguments, CompressedStorageAccountName, out storageAccountName, required: true);
 
-            string storagePath = null;
-            arguments.TryGetValue("-compressedStoragePath", out storagePath);
+                string storageKeyValue;
+                TryGetArgument(arguments, CompressedStorageKeyValue, out storageKeyValue, required: true);
 
-            StorageCredentials credentials = new StorageCredentials(storageAccountName, storageKeyValue);
-            CloudStorageAccount account = new CloudStorageAccount(credentials, true);
-            return new AzureStorageFactory(account, storageContainer, storagePath, storageBaseAddress) { Verbose = verbose, CompressContent = true };
-        }
+                string storageContainer;
+                TryGetArgument(arguments, CompressedStorageContainer, out storageContainer, required: true);
 
-        public static StorageFactory CreateStorageFactory(IDictionary<string, string> arguments, bool verbose)
-        {
-            Uri storageBaseAddress = null;
-            string storageBaseAddressStr;
-            if (arguments.TryGetValue("-storageBaseAddress", out storageBaseAddressStr))
-            {
-                storageBaseAddressStr = storageBaseAddressStr.TrimEnd('/') + "/";
+                string storagePath = null;
+                TryGetArgument(arguments, CompressedStoragePath, out storagePath);
 
-                storageBaseAddress = new Uri(storageBaseAddressStr);
+                StorageCredentials credentials = new StorageCredentials(storageAccountName, storageKeyValue);
+                CloudStorageAccount account = new CloudStorageAccount(credentials, true);
+                return new AzureStorageFactory(account, storageContainer, storagePath, storageBaseAddress) { Verbose = verbose, CompressContent = true };
             }
-
-            string storageType;
-            if (!arguments.TryGetValue("-storageType", out storageType))
+            catch (ArgumentException)
```

Comment:
It looks like a lot of places that call these functions (not specifically this one) use the null value to indicate that it should do a printUsage call. I am reluctant to change this for all of these with this changeset. We should re-evaluate how/when we should fail and when printUsage should be called for this globally.

---
In reply to: [71024082](https://github.com/NuGet/NuGet.Services.Metadata/pull/77#discussion_r71024082) [](ancestors = 71024082)